client/src/main/java/io/pravega/client/segment/impl/SegmentMetadataClientImpl.java

```diff
@@ -63,7 +64,7 @@
     public SegmentMetadataClientImpl(Segment segment, Controller controller, ConnectionPool connectionPool,
                                      String delegationToken) {
         this(segment, controller, connectionPool,
-                DelegationTokenProviderFactory.create(delegationToken, controller, segment));
+                DelegationTokenProviderFactory.create(delegationToken, controller, segment, AccessOperation.READ));
```

Comment:
SegmentMetadataCLient can truncate a Stream. Is "READ" permission enough to do this ?

Reply:
This particular constructor seems to be used for testing only. I've pushed a chance documenting that.

Reply:
since the constructor is "public", it can potentially be used in other contexts. maybe we make it package private?